judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/submission/bundle/ContestItemSubmissionResource.java

```diff
@@ -246,18 +282,24 @@ public void createItemSubmission(AuthHeader authHeader, ContestItemSubmissionDat
     public ContestantAnswerSummaryResponse getAnswerSummaryForContestant(
             AuthHeader authHeader,
             String contestJid,
-            Optional<String> userJid,
+            Optional<String> username,
             Optional<String> language) {
 
         String actorJid = actorChecker.check(authHeader);
         Contest contest = checkFound(contestStore.getContestByJid(contestJid));
         checkAllowed(submissionRoleChecker.canViewOwn(actorJid, contest));
 
         boolean canSupervise = submissionRoleChecker.canSupervise(actorJid, contest);
-        String viewedUserJid = canSupervise ? userJid.orElse(actorJid) : actorJid;
+        String viewedUserJid;
+        if (canSupervise && username.isPresent()) {
+            viewedUserJid = userSearchService.translateUsernamesToJids(ImmutableSet.of(username.get()))
```

Comment:
This can be null if the username is invalid

Reply:
Changed to `.get(...)`